### PR TITLE
Add `enable_style` option

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/2984

```console
$ lfortran a.f90
i is 10
$ lfortran a.f90 --enable-style
style suggestion: Use 'end do' instead of 'enddo'
 --> a.f90:5:5
  |
5 |     enddo
  |     ^^^^^ help: write this as 'end do'

style suggestion: Use 'end if' instead of 'endif'
 --> a.f90:8:5
  |
8 |     endif
  |     ^^^^^ help: write this as 'end if'


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
i is 10
```

Does this align with the expected outcome?

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
Not ready for review.

**Comment by gxyd** [comment]
Maybe for some of the updated tests (maybe for the unique styling test), we can have `ast_enable_style=true` in run_tests.py to ensure that styling actually works as expect it to?

File: src/libasr/diagnostics.cpp [hunk 1]
```diff
@@ -65,12 +65,17 @@ std::string Diagnostics::render(LocationManager &lm,
         const CompilerOptions &compiler_options) {
     std::string out;
     for (auto &d : this->diagnostics) {
-        if (compiler_options.no_warnings && d.level != Level::Error) {
-            continue;
-        }
         if (compiler_options.error_format == "human") {
-            out += render_diagnostic_human(d, lm, compiler_options.use_colors,
+            std::string temp = "";
+            temp += render_diagnostic_human(d, lm, compiler_options.use_colors,
```
Comment (gxyd) [inline src/libasr/diagnostics.cpp:70]: it seems like `temp` is only used in `else` block, maybe move it under there? To avoid this computation maybe.

**Comment by HarshitaKalani** [comment]
Thanks @gxyd !

**Comment by certik** [comment]
Thanks! Let's make the suggestions by default for now. We can later always change it to "no suggestion by default". But if if remove it now, we can't turn it on by default later without annoying a lot of users.

**Comment by kargl** [comment]
> Thanks! Let's make the suggestions by default for now. We can later always change it to "no suggestion by default". But if we remove it now, we can't turn it on by default later without annoying a lot of users.

@HarshitaKalani, thanks for the fix.  This will make run lfortran against my 35+ years of Fortran code much more pleasant.

 @certik, I'm fine enabling style suggestions by default.  Is the negation `--disable-style` or `--no-enable-style`?

**Comment by certik** [comment]
@kargl this was fixed in https://github.com/lfortran/lfortran/pull/3910, we added `--no-style-warnings`.

Eventually we'll have to be a lot more granular. After we reach beta, we'll rework the options so that they are all consistent. Possibly using the usual `-W*` convention.

**Comment by kargl** [comment]
Thanks.  This will make testing with older codes much more pleasant.  In my case, I had style suggestions scrolling the actual issue off the screen.

**Comment by certik** [comment]
@kargl thanks for testing LFortran. Once this PR is in: https://github.com/lfortran/lfortran/pull/3929, you could just use `--std=f23` or `--std=legacy` and that should behave exactly as you want.

**Pull request closed without merging**